File: app/src/main/java/io/github/marcinrogacki/goworld/MainActivity.java
package io.github.marcinrogacki.goworld;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ViewAnimator;
import android.view.MotionEvent;
import android.widget.TextView;
import android.widget.ProgressBar;

import com.github.pwittchen.swipe.library.Swipe;
import com.github.pwittchen.swipe.library.SwipeListener;

import io.github.marcinrogacki.goworld.Character;
import io.github.marcinrogacki.goworld.Item;

public class MainActivity extends AppCompatActivity {

    ViewAnimator viewAnimator;
    private Swipe swipe;
    private Character player;
    private Character monster;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        viewAnimator = (ViewAnimator)findViewById(R.id.viewanimator);

        swipe = new Swipe(70, 300);
        SwipeListener swipeListener = createSwipeListener();
        swipe.setListener(swipeListener);
        player = new Character();
        monster = new Character();
        updateScreens();
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent event) {
        swipe.dispatchTouchEvent(event);
        return super.dispatchTouchEvent(event);
    }

    private SwipeListener createSwipeListener() {
        return new SwipeListener() {
            @Override public void onSwipingLeft(final MotionEvent event) {
            }

            @Override public void onSwipedLeft(final MotionEvent event) {
              player.heal();
              monster.heal();
              updateScreens();
            }

            @Override public void onSwipingRight(final MotionEvent event) {
            }

            @Override public void onSwipedRight(final MotionEvent event) {
              player.attack(monster);
              monster.attack(player);
              updateScreens();
            }

            @Override public void onSwipingUp(final MotionEvent event) {
            }

            @Override public void onSwipedUp(final MotionEvent event) {
            }

            @Override public void onSwipingDown(final MotionEvent event) {
            }

            @Override public void onSwipedDown(final MotionEvent event) {
            }
        };
    }

    private void updateScreens() {
        TextView playerView = (TextView)findViewById(R.id.player);
        playerView.setText(String.valueOf("Player: " + player.getLife()));
        ProgressBar playerBar = (ProgressBar)findViewById(R.id.playerBar);
        playerBar.setProgress(player.getLife());

        TextView monsterView = (TextView)findViewById(R.id.monster);
        monsterView.setText(String.valueOf("Monster lvl1: " + monster.getLife()));
        ProgressBar monsterBar = (ProgressBar)findViewById(R.id.monsterBar);
        monsterBar.setProgress(monster.getLife());

    }
}

File: app/src/main/java/io/github/marcinrogacki/goworld/MainActivity.kt
package io.github.marcinrogacki.goworld

import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.activity.ComponentActivity
import com.github.pwittchen.swipe.library.rx2.Swipe
import com.github.pwittchen.swipe.library.rx2.SwipeListener
import kotlin.random.Random

class MainActivity : ComponentActivity() {
    private var swipe: Swipe = Swipe(70, 300)
    private var player: Character = Character()
    private var monster: Character = Character()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        swipe.setListener(createSwipeListener())
        updateScreens()
    }

    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
        swipe.dispatchTouchEvent(event)
        return super.dispatchTouchEvent(event)
    }

    private fun createSwipeListener(): SwipeListener {
        return object : SwipeListener {
            override fun onSwipingLeft(event: MotionEvent) {}
            override fun onSwipedLeft(event: MotionEvent): Boolean {
                player.heal()
                if(Random.nextBoolean()) monster.heal() else monster.attack(player)
                updateScreens()
                return true
            }

            override fun onSwipingRight(event: MotionEvent) {}
            override fun onSwipedRight(event: MotionEvent): Boolean {
                player.attack(monster)
                if(Random.nextBoolean()) monster.heal() else monster.attack(player)
                updateScreens()
                return true
            }

            override fun onSwipingUp(event: MotionEvent) {}
            override fun onSwipedUp(event: MotionEvent): Boolean { return true }
            override fun onSwipingDown(event: MotionEvent) {}
            override fun onSwipedDown(event: MotionEvent): Boolean { return true }
        }
    }

    private fun updateScreens() {
        val playerView: TextView = findViewById<View>(R.id.player) as TextView
        val playerTxt = "Marcin: " + player.lifeText
        playerView.text = playerTxt
        val playerBar: ProgressBar = findViewById<View>(R.id.playerBar) as ProgressBar
        playerBar.progress = player.life
        val monsterView: TextView = findViewById<View>(R.id.monster) as TextView
        val monsterTxt = "Rat: " + monster.lifeText
        monsterView.text = monsterTxt
        val monsterBar: ProgressBar = findViewById<View>(R.id.monsterBar) as ProgressBar
        monsterBar.progress = monster.life
    }
}
